### PR TITLE
Add compliant and noncompliant examples of javascript/sql-injection@v1.0

### PR DESCRIPTION
**Pull request by ankit-amazon**
*Description of changes:*
* Rules Added: javascript/sql-injection@v1.0
* Added new functions for compliant and non compliant examples
* Added .gitignore

### CODE REVIEW

File: src/javascript/detectors/sql_injection/sql_injection.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ */
+
+// {fact rule=sql-injection@v1.0 defects=1}
+var sql = require('mysql')
+var express = require('express')
+var app = express()
+
+var connection = sql.createConnection({
+    host     : 'localhost',
+    user     : 'newUserName',
```
Comment (ctlai95) [inline src/javascript/detectors/sql_injection/sql_injection.js:13]: nit: can we make this `myUserName` for consistency

Comment (ankit-amazon) [reply src/javascript/detectors/sql_injection/sql_injection.js:13]: sure

File: src/javascript/detectors/sql_injection/sql_injection.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    host     : 'localhost',
+    user     : 'myUserName',
+    password : 'myPass',
+    database : 'myDatabase'
+})
+
+function sqlInjectionCompliant() {
+    app.get("/user/:id", (req, res) => {
+        var query = "SELECT * FROM Employees WHERE ID = " + connection.escape(req.params.id)
+        // Compliant: user input is sanitized before use.
```
Comment (mj632) [inline src/javascript/detectors/sql_injection/sql_injection.js:45]: nit: this comment should go before line `44` (the line above  `var query = "SELECT * FROM Employees WHERE ID = " + connection.escape(req.params.id)`)

Comment (ankit-amazon) [reply src/javascript/detectors/sql_injection/sql_injection.js:44]: Improved.

**Review by ctlai95 — APPROVED** [review]

**Pull request merged**